spec/generators/sorbet_test_cases.rb

```diff
@@ -114,48 +186,38 @@
 T.assert_type!(spell_books_query.find_by!(name: 'Fantastic Beasts'), SpellBook)
 T.assert_type!(spell_books_query.find_by_id(spell_book.id), T.nilable(SpellBook))
 T.assert_type!(spell_books_query.find_by_id!(spell_book.id), SpellBook)
+T.assert_type!(spell_books_query.find_or_initialize_by(name: 'Fantastic Beasts'), SpellBook)
+T.assert_type!(spell_books_query.find_or_create_by(name: 'Fantastic Beasts'), SpellBook)
+T.assert_type!(spell_books_query.find_or_create_by!(name: 'Fantastic Beasts'), SpellBook)
+# T.assert_type!(spell_books_query.new, SpellBook) # Ignored until we add support
+# T.assert_type!(spell_books_query.build, SpellBook) # Ignored until we add support # Ignored until we add support
+# T.assert_type!(spell_books_query.create(name: 'Fantastic Beasts'), SpellBook) # Ignored until we add support
+# T.assert_type!(spell_books_query.create!(name: 'Fantastic Beasts'), SpellBook) # Ignored until we add support
+# T.assert_type!(spell_books_query.first_or_create(name: 'Fantastic Beasts'), SpellBook) # Ignored until we add support
+# T.assert_type!(spell_books_query.first_or_create!(name: 'Fantastic Beasts'), SpellBook) # Ignored until we add support
+# T.assert_type!(spell_books_query.first_or_initialize, SpellBook) # Ignored until we add support
+spell_books_query.find_each { |s| T.assert_type!(s, SpellBook) }
+T.assert_type!(spell_books_query.find_each, T::Enumerator[SpellBook])
+spell_books_query.find_in_batches { |s| T.assert_type!(s, T::Array[SpellBook]) }
+T.assert_type!(spell_books_query.find_in_batches, T::Enumerator[T::Array[SpellBook]])
+# T.assert_type!(spell_books_query.destroy_all, T::Array[SpellBook]) # Ignored until we add support
+T.assert_type!(spell_books_query.any?, T::Boolean)
+T.assert_type!(spell_books_query.many?, T::Boolean)
+T.assert_type!(spell_books_query.none?, T::Boolean)
+T.assert_type!(spell_books_query.one?, T::Boolean)
+# T.assert_type!(spell_books_query.update_all(name: 'Fantastic Beasts'), Integer) # Ignored until we add support
+# T.assert_type!(spell_books_query.delete_all, Integer) # Ignored until we add support
 # Query chaining
 T.assert_type!(spell_books_query.preload(:wizard), SpellBook::ActiveRecord_AssociationRelation)
 T.assert_type!(spell_books_query.eager_load(:wizard), SpellBook::ActiveRecord_AssociationRelation)
 T.assert_type!(spell_books_query.order(:id), SpellBook::ActiveRecord_AssociationRelation)
 T.assert_type!(spell_books_query.where.not(id: 1), SpellBook::ActiveRecord_AssociationRelation)
 T.assert_type!(spell_books_query.biology, SpellBook::ActiveRecord_AssociationRelation)
-
-# Enumerable on activerecord relation
-T.assert_type!(Wizard.all.to_a, T::Array[Wizard])
-Wizard.all.each do |w|
-  T.assert_type!(w, Wizard)
-end
-Wizard.all.map do |w|
-  T.assert_type!(w, Wizard)
-end
-Wizard.all.to_a.map do |w|
```

Comment:
This assertion wasn't super useful since we already had an assertion testing that `.to_a` would return a `T::Array[Wizard]`.

Reply:
The first one is useful because sometimes the "map" or the "each" function is overridden by Sorbet autogen rbi. I have it here to make sure that "each" and "map" work correctly.

Reply:
Some of them, like the check `Wizard.all.map` are there to also make sure that sorbet autogen rbi doesn't make it stop working for subclasses.

However, I see you also have these checks above. We can remove them here.

Reply:
Yea I kept those but moved them into the sections for relation, association relation, and collection association. I just removed this one test for them since we were already ensuring that to_a returned an array of things, so we shouldn’t need to test that to_a.each has the correct behavior unless we’re skeptical of sorbet’s sigs for builtin ruby methods.